### PR TITLE
Dataset geo/date multipart annotation form fix

### PR DESCRIPTION
**Pull request by adamgilg**
https://github.com/jataware/dojo/issues/172

Both the geo & date multipart forms were looking bad (as was the model templater) because the labels weren't shrinking (becoming small at the top of the input). The label ended up overlaid on top of the initially selected input content, making it difficult to read. 

All the uses of FormikSelect - the Templater and both multipart forms - were not showing any `required` errors. This PR also fixes that. 

Screenshot 2023-10-06 at 11 55 21 AM

### CODE REVIEW

**Comment by ccjoel** [comment]
Found a  very edgey and interesting edge case:

[Screencast from 2023-10-09 11-10-32.webm](https://github.com/jataware/dojo/assets/1967061/c1a8a896-4fc3-4022-b3b6-41fa455d62da)

Users may break the app if they explore too much. I've never seen a recursive/loop on this page/form/formik/mui like this. I wonder if the maximum callstack error was caused by the recent changes + changes on this PR.

I was able to reproduce consistently following exactly that order of operations from the video.

@adamgilg I'll let it up to you on if you think this is worth pursuing.

**Comment by ccjoel** [comment]
I forgot to mention- this looks great btw. All of it went back to how it looked before the mui upgrade: the alignments of the fields, the alignment of the qualifies, the label shrink property 💯 !

File: ui/client/components/formikComponents/FormikSelect.js [hunk 1]
```diff
@@ -3,9 +3,21 @@ import { useField, useFormikContext } from 'formik';
 
 import InputLabel from '@mui/material/InputLabel';
 import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
 import MenuItem from '@mui/material/MenuItem';
 import Select from '@mui/material/Select';
 
+// removes the [' '] from dataset multicolumn date & geo, which are there
```
Comment (ccjoel) [inline ui/client/components/formikComponents/FormikSelect.js:10]: Great comment. I was trying to find evidence of this workaround online, and I did find an old github issue that illustrates this. Will leave here in the PR for reference:

https://github.com/jaredpalmer/formik/issues/2278

**Comment by ccjoel** [comment]
Tested it manually to ensure:
- Unit tests pass (build-a-get data is transform as it should). All UI jest unit tests passed.
- Checked that the input of the processing fn still works, and that all 3 portions of the build-a-date are sent to backend on next/save to Transformation page (overlaps somewhat with the unit test validity).

If we feel we can let the edge case I found for later, we can file an issue for it. Would be nice to address it- maybe timebox it? Let me know if you need any help with that. Thanks!

**Comment by adamgilg** [comment]
> Found a very edgey and interesting edge case:
> Screencast.from.2023-10-09.11-10-32.webm
> 
> Users may break the app if they explore too much. I've never seen a recursive/loop on this page/form/formik/mui like this. I wonder if the maximum callstack error was caused by the recent changes + changes on this PR.
> 
> I was able to reproduce consistently following exactly that order of operations from the video.
> 
> @adamgilg I'll let it up to you on if you think this is worth pursuing.

I just checked, this bug also exists on `main` (that is, it isn't related to this fix). The issue, as I see it, is twofold: 

1. We're allowing users to select columns that have already been annotated in the various dropdowns for multicolumn annotations. Presumably something can't be annotated twice, and if it can, we'll need to ensure that this actually works. 
2. We need to make the date `FormatValidationInput` check run again when the select changes. That you're able to change the value and it doesn't display an error also contributes to this (and also exists on `main`). 

If we fixed both of these things, this wouldn't be possible. It is quite an edge case, however, so I don't really see it being worth the effort right now. But definitely something to file an issue for.

**Comment by adamgilg** [comment]
> I forgot to mention- this looks great btw. All of it went back to how it looked before the mui upgrade: the alignments of the fields, the alignment of the qualifies, the label shrink property 💯 !

And the fields show errors now, which they didn't before!

**Comment by adamgilg** [comment]
@ccjoel I opened https://github.com/jataware/dojo/issues/176 & https://github.com/jataware/dojo/issues/177 about the above.

**Review by ccjoel — APPROVED** [review]
🚢 ➡️ 🌎 ❗
